### PR TITLE
[autoscaler] hotfix calculate_node_resources

### LINKED ISSUE #10869: [autoscaler] Inconsistent reporting of worker node types
On master, it seems I have a cluster with a cpu_4_ondemand head node and cpu_64_ondemand worker (68 CPUs in total), but somehow the debug string is reporting 2 workers of cpu_64_ondemand (which is wrong, since there is only 1 worker node, and the head node is cpu_4_ondemand).

YAML: https://gist.github.com/ericl/f9c32b8bbbe4e1aed43899832730f70d (switching the Ray version to latest wheel)

```
 - ResourceUsage: 0.0/68.0 CPU, 0.0/1.0 Custom1, 0.0 GiB/198.65 GiB memory, 0.0
GiB/59.19 GiB object_store_memory
 - TimeSinceLastHeartbeat: Min=0 Mean=0 Max=0
Worker node types:
 - cpu_64_ondemand: 2

```

cc @wuisawesome

### PR DESCRIPTION
**Pull request by wuisawesome**
## Related issue number

closes #10869

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/resource_demand_scheduler.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ def get_nodes_to_launch(self, nodes: List[NodeID],
             nodes: List of existing nodes in the cluster.
             pending_nodes: Summary of node types currently being launched.
             resource_demands: Vector of resource demands from the scheduler.
+            usage_by_ip: Mapping from ip to available resources.
```
Comment (wuisawesome) [inline python/ray/autoscaler/resource_demand_scheduler.py:59]: incidental (should've documented this a while ago)

File: python/ray/autoscaler/resource_demand_scheduler.py [hunk 2]
```diff
@@ -90,14 +91,9 @@ def calculate_node_resources(
...
             if node_type not in self.node_types:
-                logger.warn(
-                    f"Missing entry for node_type {node_type} in "
-                    f"cluster config: {self.node_types} under entry "
-                    f"available_node_types. This node's resources will be "
-                    f"ignored. If you are using an unmanaged node, manually "
-                    f"set the user_node_type tag to \"{NODE_KIND_UNMANAGED}\""
-                    f"in your cloud provider's management console.")
-                return None
+                raise RuntimeError("Missing entry for node_type {} in "
```
Comment (ericl) [inline python/ray/autoscaler/resource_demand_scheduler.py:94]: Was this doing something broken before?

**Review by ericl — APPROVED** [review]

**Comment by ericl** [comment]
cc @barakmich we should hotfix this

File: python/ray/autoscaler/resource_demand_scheduler.py (hunk 2 above)
Comment (wuisawesome) [reply python/ray/autoscaler/resource_demand_scheduler.py]: just reverted it, no clue how that happened

File: python/ray/autoscaler/resource_demand_scheduler.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 from typing import List, Dict, Tuple
 
 from ray.autoscaler.node_provider import NodeProvider
-from ray.autoscaler.tags import TAG_RAY_USER_NODE_TYPE, NODE_KIND_UNMANAGED
+from ray.autoscaler.tags import TAG_RAY_USER_NODE_TYPE
```
Comment (ericl) [inline python/ray/autoscaler/resource_demand_scheduler.py:17]: import should be fixed

**Pull request merged**